libraries/eosiolib/core/eosio/crypto.hpp

```diff
@@ -276,6 +276,49 @@ namespace eosio {
     */
    void assert_ripemd160( const char* data, uint32_t length, const eosio::checksum160& hash );
 
+   /**
+    *  Hashes `data` using SHA3 NIST.
+    *
+    *  @ingroup crypto
+    *  @param data - Data you want to hash
+    *  @param length - Data length
+    *  @return eosio::checksum256 - Computed digest
+    */
+   eosio::checksum256 sha3( const char* data, uint32_t length );
+
+   /**
+    *  Tests if the SHA3 hash generated from data matches the provided digest.
+    *
+    *  @ingroup crypto
+    *  @param data - Data you want to hash
+    *  @param length - Data length
+    *  @param hash - digest to compare to
+    *  @note This method is optimized to a NO-OP when in fast evaluation mode.
+    */
+   void assert_sha3( const char* data, uint32_t length, const eosio::checksum256& hash );
```

Comment:
Should we really have this?

It isn't hard for the user to do this check themselves using the `sha3` function and that way they get to provide their own error message.

Also, this can mislead the developer to think that this is calling a host function which can be optimized away in replay, which it is not. In fact, the `@note` on this function that you copied is incorrect.

Reply:
Yeah, I would rather keep these as they have good symmetry with the other crypto hashing functions.  But yes I will change the note to be explicit.